src/extension/wps/wps-core/src/main/java/org/geoserver/wps/executor/WPSExecutionManager.java

```diff
@@ -391,6 +391,13 @@ private ExecuteResponseType execute() {
                 SubProgressListener executionListener = new SubProgressListener(listener,
                         inputPercentage, executionPercentage);
                 notifier.checkDismissed();
+                
+                //The system only moves from Queued to Running when the progress notifier fires (!!!).  This fires when inputs are read.
+                //   However, if there are no inputs, it never moves to running state!
+                //Since we are executing, we can move this from queued to running state. 
+                if (status.getPhase() == ProcessState.QUEUED) {
```

Comment:
I think I understand, I considered recommending a "first" input parameter that is evaluated but not included in the list, so a no argument process would only have the one entry. It seemed a bit of a hack.